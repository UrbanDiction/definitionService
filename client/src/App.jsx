/* eslint-disable */
import React from "react";
import DefinitionList from "./components/DefinitionList.jsx";

const fetch = require("node-fetch");

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      data: []
    };
  }
  // eslint-disable-next-line class-methods-use-this
  componentDidMount() {
    fetch("http://localhost:8001/definition/word", {
      method: "POST",
      body: JSON.stringify({ word: "architecto" }),
      headers: {
        "Content-Type": "application/json"
      }
    })
      .then(data => {
        // console.log(data);
        return data.json();
      })
      .then(({ definitionQuery }) => {
        this.setState({ data: definitionQuery });
      });
  }
  // eslint-disable-next-line
  render() {
    return (
      <div>


      <div>
        <DefinitionList data={this.state.data} />
      </div>
      </div>
    );
  }
}

export default App;
