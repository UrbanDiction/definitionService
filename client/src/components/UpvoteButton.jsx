/* eslint-disable */
import React from "react";

class UpvoteButton extends React.Component {
  constructor() {
    super();
  }
  render() {
    return (
      <div>UpvoteTest</div>
    )
  }
}

export default UpvoteButton;