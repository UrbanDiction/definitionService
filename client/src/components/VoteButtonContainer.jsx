/* eslint-disable */
import React from "react";
import UpvoteButton from "./UpvoteButton.jsx";
import DownvoteButton from "./DownvoteButton.jsx";

class VoteButtonContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      buttonToggle: "none",
      upvotes: this.props.upvotes,
      downvotes: this.props.downvotes
    };
    this.toggleButton = this.toggleButton.bind(this);
  }

  toggleButton(button) {
    if (this.state.buttonToggle === button) {
      return;
    }
    if (this.state.buttonToggle != "none") {
      let newUpvotes = this.state.upvotes, newDownvotes = this.state.downvotes;
      if (button === "upvote") {
        newDownvotes = this.state.downvotes - 1;
        newUpvotes = this.state.upvotes + 1;
      } else {
        newDownvotes = this.state.downvotes + 1;
        newUpvotes = this.state.upvotes - 1;
      }
      this.setState({buttonToggle: button, upvotes: newUpvotes, downvotes: newDownvotes})
    } else {
      let newUpvotes = this.state.upvotes, newDownvotes = this.state.downvotes;
      if (button === "upvote") {
        newUpvotes = this.state.upvotes + 1;
      } else {
        newDownvotes = this.state.downvotes + 1;
      }
      this.setState({buttonToggle: button, upvotes: newUpvotes, downvotes: newDownvotes})

    }
  }

  render() {
    return (
      <div className="btn-group">
        <UpvoteButton
          toggleButton={this.toggleButton}
          buttonState={this.state.buttonToggle}
          upvotes={this.state.upvotes}
        />
        <DownvoteButton
          toggleButton={this.toggleButton}
          buttonState={this.state.buttonToggle}
          downvotes={this.state.downvotes}
        />
      </div>
    );
  }
}

export default VoteButtonContainer;
