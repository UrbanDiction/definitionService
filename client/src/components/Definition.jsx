/* eslint-disable */
import React from "react"; // eslint-disable-line
import VoteButtonContainer from "./VoteButtonContainer.jsx";

class Definition extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      createdBy: this.props.data.created_by,
      createdDate: this.props.data.created_date,
      definition: this.props.data.definition,
      example: this.props.data.example,
      upVotes: this.props.data.upvotes,
      downVotes: this.props.data.downvotes,
      hashtags: this.props.data.hash_tags,
      id: this.props.data.id,
      word_id: this.props.data.word_id
    };
  }
  render() {
    return (
      <div>
        <VoteButtonContainer
          upvotes={this.state.upVotes}
          downvotes={this.state.downVotes}
        />
      </div>
    );
  }
}

export default Definition;
