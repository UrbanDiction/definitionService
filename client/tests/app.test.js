/* eslint-disable */
import React from "react";
import { shallow, mount } from "enzyme"; // mount removed for renderer
//import renderer from "react-test-renderer";
import ReactTestUtils from 'react-dom/test-utils';
import App from "../src/App.jsx";
import Definition from "../src/components/Definition.jsx"
import VoteButtonContainer from "../src/components/VoteButtonContainer.jsx"
import UpvoteButton from "../src/components/UpvoteButton.jsx"

it("works", () => {
  expect(App).toEqual(App);
});

describe("First React component test with Enzyme", () => {
  it("renders without crashing", () => {
  const app = shallow(<App />);
  expect(app).toEqual(shallow(<App />))
  });
});



test('Definition component renders text of definition', () => {
  const testData = {
    created_by: "Nick",
    created_date: "2017-08-02",
    definition: "test def",
    downvotes: 1,
    example: "test example",
    hash_tags: "test",
    id: 1,
    upvotes: 1,
    word_id: 1
  }
  const wrapper = mount(
    <Definition data={testData} />
  );

  const p = wrapper.find('.single_def');

  expect(p.text()).toBe("test def")
})

// describe('VoteButtonContainer component renders the buttons correctly', () => {
//   it('renders correctly', () => {
//     const defData = {
//       created_by: "Nick",
//       created_date: "2017-08-02",
//       definition: "test def",
//       downvotes: 1,
//       example: "test example",
//       hash_tags: "test",
//       id: 1,
//       upvotes: 1,
//       word_id: 1
//     }
//     const rendered = renderer.create(
//       <VoteButtonContainer upvotes={defData.upvotes} downvotes={defData.downvotes} />
//     );
//     expect(rendered.toJSON()).toMatchSnapshot();
//   });
// });

// test('Definition component renders text of definition creator', () => {
//   const testData = {
//     created_by: "Nick",
//     created_date: "2017-08-02",
//     definition: "test def",
//     downvotes: 1,
//     example: "test example",
//     hash_tags: "test",
//     id: 1,
//     upvotes: 1,
//     word_id: 1
//   }
//   const wrapper = mount(
//     <Definition data={testData} />
//   );

//   const p = wrapper.find('.author');

//   expect(p.text()).toBe("Nick")
// })



test('VoteButtonContainer toggles buttonToggle property when vote button is clicked', () => {
  const toggleButton = jest.fn();
  const testData = {
    created_by: "Nick",
    created_date: "2017-08-02",
    definition: "test def",
    downvotes: 1,
    example: "test example",
    hash_tags: "test",
    id: 1,
    upvotes: 1,
    word_id: 1
  }
  //  const rendered = renderer.create(
  //    <VoteButtonContainer data={testData}  />
  //  );

  const wrapper = mount(
    <Definition data={testData} toggleButton={toggleButton} />
  );
  let p = wrapper.find('UpvoteButton');
  p.simulate('click')
  p = wrapper.find('DownvoteButton')
  p.simulate('click')
})

// test('VoteButtonContainer component renders number of upvotes', () => {
//   const testData = {
//     created_by: "Nick",
//     created_date: "2017-08-02",
//     definition: "test def",
//     downvotes: 1,
//     example: "test example",
//     hash_tags: "test",
//     id: 1,
//     upvotes: 1,
//     word_id: 1
//   }
//   const wrapper = mount(
//     <UpvoteButton data={testData} />
//   );

//   const div = wrapper.find('.upvotes');
//   expect(div.text()).to.equal(1)
// })


















