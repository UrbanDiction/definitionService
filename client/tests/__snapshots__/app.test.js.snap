// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition component renders the definition correctly renders correctly 1`] = `
<div
  className="defs"
>
  <div
    className="top-row"
  >
    <div
      className="def-count"
    />
    <div
      className="social-media"
    >
      <div
        className="twitter"
      >
        <a
          href="https://www.twitter.com/"
          target="_blank"
        >
          <i>
            <svg
              className="svg-twitter"
              viewBox="0 0 34 34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.3 8c5.4 2.8 9.9 2.6 9.9 2.6s-1.7-6.2 3.6-8.9 9 1.9 9 1.9.9-.3 1.6-.5c.7-.3 1.7-.7 1.7-.7l-1.6 3 2.5-.3s-.3.5-1.3 1.4l-1.4 1.4s.4 7.4-3.4 13.1c-3.8 5.7-8.7 9.1-15.9 9.9C4.9 31.6.2 28.6.2 28.6s3.1-.2 5.1-1 4.9-2.8 4.9-2.8-4.1-1.3-5.5-2.7c-1.5-1.4-1.8-2.3-1.8-2.3l4-.1s-4.2-2.3-5.4-4.1S0 12.1 0 12.1l3.1 1.3S.5 9.8.2 7s.5-4.3.5-4.3S1.9 5.2 7.3 8z"
              />
            </svg>
          </i>
        </a>
      </div>
      <div
        className="facebook"
      >
        <a
          href="https://www.facebook.com/"
          target="_blank"
        >
          <i>
            <svg
              className="svg-facebook"
              viewBox="0 0 34 34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19.4 0c-1 0-2 .1-3.2.6-2.3 1-3.5 3.3-3.5 6.9V11H9.4v6h3.3v17h6.7V17h4.5l.6-5.9h-5.2V8.4c0-.8.1-1.4.3-1.8.3-.5.9-.8 1.8-.8h3V0h-5z"
              />
            </svg>
          </i>
        </a>
      </div>
      <div
        className="share"
      >
        <i>
          <svg
            className="svg-share"
            viewBox="0 0 34 34"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M27.1 12c-1.3 0-2.5.5-3.4 1.4l-11.9-7c.2-.3.2-.9.2-1.2.1-3-2.1-5.2-5-5.2S2 2.2 2 5.1s2.2 5.1 5 5.1c1.3 0 2.5-.5 3.4-1.4l11.9 7c-.2.3-.2.9-.2 1.2 0 .3 0 .9.2 1.2l-11.9 7.2c-.8-.9-2-1.4-3.4-1.4-2.7 0-4.9 2.2-4.9 5s2.2 5 4.9 5 4.9-2.2 4.9-5c0-.3 0-.7-.2-1.2l11.9-7.2c.8.9 2 1.4 3.4 1.4 2.8 0 5-2.2 5-5.1S30 12 27.1 12z"
            />
          </svg>
        </i>
      </div>
    </div>
  </div>
  <div
    className="word_title"
  >
    Word
  </div>
  <p
    className="single_def"
  >
    test def
  </p>
  <p
    className="example"
  >
    test example
  </p>
  <div
    className="hash"
  >
    <span
      className="hash_green"
    >
      #test 
    </span>
  </div>
  <div
    className="author-date"
  >
    by 
    <span
      className="author"
    >
      Nick
    </span>
     
    2017-08-02
  </div>
  <div
    className="buttonRow"
  >
    <div
      className="btn-group"
    >
      <div
        className="upvote-wrapper"
        onClick={[Function]}
      >
        <div
          className="upvote-img"
        >
          <i>
            <svg
              viewBox="0 0 34 34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M34 15.3c0-1.9-1.4-3.4-3.1-3.4h-9.7l1.5-7.8v-.5c0-.7-.3-1.4-.6-1.9L20.4 0 10.2 11.2c-.6.5-.9 1.4-.9 2.4v17c0 1.9 1.4 3.4 3.1 3.4h13.9c1.2 0 2.3-.8 2.8-2l4.6-12.1c.2-.3.2-.9.2-1.2v-3.4h.1c0 .2 0 0 0 0zM0 34h6.2V13.6H0V34z"
              />
            </svg>
          </i>
        </div>
        <div
          className="upvotes"
        >
          1
        </div>
      </div>
      <div
        className="downvote-wrapper"
        onClick={[Function]}
      >
        <div
          className="downvote-img"
        >
          <i>
            <svg
              viewBox="0 0 34 34"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M27.8 0v20.4H34V0h-6.2zm-6.2 0H7.7C6.5 0 5.4.9 4.9 2L.3 14.1c-.1.3-.3.7-.3 1.2v3.4c0 1.9 1.4 3.4 3.1 3.4h9.7l-1.5 7.8v.5c0 .7.3 1.4.6 1.9l1.7 1.7 10.2-11.2c.6-.7.9-1.5.9-2.4v-17c0-1.9-1.4-3.4-3.1-3.4z"
              />
            </svg>
          </i>
        </div>
        <div
          className="downvotes"
        >
          1
        </div>
      </div>
    </div>
    <div
      className="flag-elip"
    >
      <div
        className="flag-button"
      >
        <i>
          <svg
            id="Capa_1"
            version="1.1"
            viewBox="0 0 496.825 496.825"
            width="16px "
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <g>
              <path
                d="M248.413,18.7c-95.625-42.075-153,0-153,0v124.312v353.812h19.125V219.513c24.862-9.562,70.763-17.212,133.875,9.562 c95.625,42.075,153,0,153,0V18.7C401.413,18.7,344.038,60.775,248.413,18.7z"
              />
            </g>
          </svg>
        </i>
      </div>
      <div
        className="elip-button"
      >
        ...
      </div>
    </div>
  </div>
  <div
    className="tone"
    style={
      Object {
        "backgroundColor": "#efff00",
      }
    }
  >
    <div
      className="mug-text"
      style={
        Object {
          "color": "#000000",
        }
      }
    >
      Get a 
      <span
        className="mug-word"
        style={
          Object {
            "backgroundColor": "#000000",
            "color": "#efff00",
          }
        }
      >
        Word
      </span>
       mug for your cousin Steve.
    </div>
    <div
      className="mug"
    >
      <img
        className="mug-image"
        src="https://d2gatte9o95jao.cloudfront.net/assets/mug-ad-02@2x-5e65e4fe0daf91156b197594c58445606b861fc9317ca6c02fad7b75dfb58e22.png"
      />
    </div>
  </div>
</div>
`;
